pkg/operator/podidentity/gcppodidentitywebhook.go

```diff
@@ -32,3 +35,8 @@ func (a GcpPodIdentity) GetImagePullSpec() string {
 func (a GcpPodIdentity) Name() string {
 	return "gcp"
 }
+
+func (a GcpPodIdentity) ApplyDeploymentSubstitutionsInPlace(deployment *appsv1.Deployment, client client.Client, logger log.FieldLogger) error {
+	deployment.Spec.Template.Spec.Containers[0].Image = a.GetImagePullSpec()
```

Comment:
This is required for all platforms. Keep this in the controller code to reduce duplicate code. Keep these platform specific functions for platform specific changes.

Reply:
yeah, it's in controller now.